src/integrator.jl

```diff
@@ -6,34 +6,36 @@
 abstract type AbstractIntegrator end
 
 struct Leapfrog{T<:AbstractFloat} <: AbstractIntegrator
-    ϵ   ::  T
+    ϵ       ::  T
+    jitter  :: T
 end
-
 Base.show(io::IO, l::Leapfrog) = print(io, "Leapfrog(ϵ=$(round(l.ϵ; sigdigits=3)))")
 
+Leapfrog(ϵ::T) where {T<:AbstractFloat} = Leapfrog(ϵ, zero(T))
+
 # Create a `Leapfrog` with a new `ϵ`
-function (::Leapfrog)(ϵ::AbstractFloat)
-    return Leapfrog(ϵ)
-end
+(lf::Leapfrog)(ϵ::AbstractFloat) = Leapfrog(ϵ, lf.jitter)
 
 # TODO: double check the function below to see if it is type stable or not
 function step(
     lf::Leapfrog{F},
     h::Hamiltonian,
     z::PhasePoint,
     n_steps::Int=1;
-    fwd = n_steps > 0 # Simulate hamiltonian backward when n_steps < 0
+    fwd::Bool=n_steps > 0 # simulate hamiltonian backward when n_steps < 0,
 ) where {F<:AbstractFloat,T<:Real}
     @unpack θ, r = z
     ϵ = fwd ? lf.ϵ : -lf.ϵ
+    iszero(lf.jitter) || (ϵ *= (1 + lf.jitter * (2 * rand() - 1)))
```

Comment:
@yebai Step size jittering is simply implemented by adding a field to the `Leapfrog` integrator and this line.